lemmy_structs/src/report.rs

```diff
@@ -0,0 +1,57 @@
+use lemmy_db::{comment_report::CommentReportView, post_report::PostReportView};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize)]
+pub struct CreateReport {
+    pub report_type: String,
+    pub entity_id: i32,
+    pub reason: String,
+    pub auth: String,
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+pub struct CreateReportResponse {
+    pub success: bool,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ListReports {
+    pub page: Option<i64>,
+    pub limit: Option<i64>,
+    pub community: i32,
```

Comment:
One thing I forgot here: the community should be optional on all these. A user should be able to view the reports for all the communities they mod.